lab_3/main.py

```diff
@@ -17,38 +20,74 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
+    is_text_incorrect = not isinstance(text, str) or not text
+    if is_text_incorrect:
+        return ()
+
+    text = text.lower().split('. ')
+    for index_sent, sentence in enumerate(text):
+        sentence = re.sub(r'[^\w\s]', '', sentence).split()
+        if sentence:
+            for index_word, word in enumerate(sentence):
+                word = '_' + word + '_'
+                sentence[index_word] = tuple(word)
+            text[index_sent] = tuple(sentence)
+            continue
+        del text[index_sent]
```

Comment:
As I rewrite the same list, strings which do not correspond _if sentence:_ will stay in list and then stay in final tuple - that's why I remove them. If I try to do this function without _if sentence:_, final tuple may contain empty tuples because if a string is without any character, it will be an empty string (as symbols are replaced with '' by regular expressions).

Reply:
I meant the `del text[index_sent]` part.

Reply:
Because after removing unnecessary symbols possible empty strings will stay in the list (and then in the tuple) if I do not delete them